packages/api/lambdas/sf-event-sqs-to-db-records.js

```diff
@@ -82,52 +120,132 @@ const shouldWriteExecutionToRDS = async (cumulusMessage, knex) => {
+  cumulusMessage,
+  collection,
+  provider,
+  knex,
+  pdrModel = new Pdr(),
+}) => {
+  // If there is no PDR in the message, then there's nothing to do here, which is fine
+  if (!messageHasPdr(cumulusMessage)) {
+    return undefined;
+  }
+  if (!isRecordDefined(collection)) {
+    throw new Error(`Collection reference is required for a PDR, got ${collection}`);
+  }
+  if (!isRecordDefined(provider)) {
+    throw new Error(`Provider reference is required for a PDR, got ${provider}`);
   }
+  return knex.transaction(async (trx) => {
+    const [cumulusId] = await writePdrViaTransaction({ cumulusMessage, collection, provider, trx });
+    await pdrModel.storePdrFromCumulusMessage(cumulusMessage);
+    return cumulusId;
+  });
 };
 
-const savePdrToDb = async (cumulusMessage) => {
-  const pdrModel = new Pdr();
+const saveGranulesToDb = async (cumulusMessage) => {
+  const granuleModel = new Granule();
+
   try {
-    await pdrModel.storePdrFromCumulusMessage(cumulusMessage);
+    await granuleModel.storeGranulesFromCumulusMessage(cumulusMessage);
   } catch (error) {
     const executionArn = getMessageExecutionArn(cumulusMessage);
-    log.fatal(`Failed to create/update PDR database record for execution ${executionArn}: ${error.message}`);
+    log.fatal(`Failed to create/update granule records for execution ${executionArn}: ${error.message}`);
     throw error;
   }
 };
 
-const saveGranulesToDb = async (cumulusMessage) => {
+const writeRecordsToDynamoDb = async (cumulusMessage) => {
+  const executionModel = new Execution();
+  const pdrModel = new Pdr();
   const granuleModel = new Granule();
 
+  const results = await Promise.allSettled([
+    executionModel.storeExecutionFromCumulusMessage(cumulusMessage),
+    pdrModel.storePdrFromCumulusMessage(cumulusMessage),
+    granuleModel.storeGranulesFromCumulusMessage(cumulusMessage),
+  ]);
+  const failures = results.filter((result) => result.status === 'rejected');
+  if (failures.length > 0) {
+    const allFailures = failures.map((failure) => failure.reason);
+    const aggregateError = new AggregateError(allFailures);
+    log.error('Failed writing some records to Dynamo', aggregateError);
+    throw aggregateError;
+  }
+  return results;
+};
+
+const writeRecords = async (cumulusMessage, knex) => {
+  const executionArn = getMessageExecutionArn(cumulusMessage);
+
+  const collection = await getMessageCollection(cumulusMessage, knex);
+  const isExecutionRDSWriteEnabled = await shouldWriteExecutionToRDS(
+    cumulusMessage,
+    collection,
+    knex
+  );
+
+  // If execution is not written to RDS, then PDRs/granules which reference
+  // execution should not be written to RDS either
+  if (!isExecutionRDSWriteEnabled) {
+    return writeRecordsToDynamoDb(cumulusMessage);
+  }
+
+  const provider = await getMessageProvider(cumulusMessage, knex);
+
   try {
-    await granuleModel.storeGranulesFromCumulusMessage(cumulusMessage);
+    console.log('about to write execution');
```

Comment:
I realize this is in-progress work/might be useful debugging, but we should remove these before 2228/make them logger statements before the work is complete.